integration-cli/docker_cli_inspect_test.go

```diff
@@ -142,33 +121,26 @@ func (s *DockerSuite) TestInspectTypeFlagWithInvalidValue(c *check.C) {
 	dockerCmd(c, "run", "--name=busybox", "-d", "busybox", "true")
 
 	out, exitCode, err := dockerCmdWithError("inspect", "--type=foobar", "busybox")
-	if exitCode != 0 || err != nil {
-		if !strings.Contains(out, "not a valid value for --type") {
-			c.Fatalf("failed to inspect image: %s, %v", out, err)
-		}
-	}
+	c.Assert(err, checker.NotNil, check.Commentf("%s", exitCode))
+	c.Assert(exitCode, checker.Equals, 1, check.Commentf("%s", err))
```

Comment:
Based on the name of the test and on I would say that the command should always return an error, so your assert are ok to me :wink: